lib/tasks/whitehall_importer.rb

```diff
@@ -50,10 +51,13 @@ def log_error(error)
 
     def create_users(users)
       users.each do |user|
-        next if User.exists?(uid: user["uid"])
-
         user_keys = %w[uid name email organisation_slug organisation_content_id]
-        User.create!(user.slice(*user_keys).merge("permissions" => []))
+        content_publisher_user = User.create_with(user.slice(*user_keys).merge("permissions" => [])).find_or_create_by!(uid: user["uid"])
```

Comment:
I found this a bit tricky to parse. I wonder if more verbose would be better?

```ruby
uid = user["uid"]
content_publisher_user = User.exists?(uid) ? User.find(uid) : create_user(user_keys)
```

Reply:
This would have to be written as follows, which seems lengthier

```
content_publisher_user = User.exists?(uid: user["uid"]) ? User.find_by(uid: user["uid"]) : User.create!(user.slice(*user_keys).merge("permissions" => []))
```

Additionally, if the user exists we'd be executing two `SELECT` queries, instead of just the one.

Reply:
I think the `User.exists?(uid)` syntax would work ok (tried it in app console on integration)... does it not work locally?

Good point about the multiple `SELECT` queries. But I think this would work via index rather than select, so should be very fast.

Reply:
(Not a blocker, mind - do feel free to merge following Kevin's "✅")

Reply:
This matches on the primary key (i.e. `id` for the users table).  It seems there's some magic going on where the first integer section of the string is being used to run the query, e.g.

```
irb(main):368:0> User.exists?("7a2917c0-bbf6-0135-c82a-1234")
  User Exists (1.6ms)  SELECT  1 AS one FROM "users" WHERE "users"."id" = $1 LIMIT $2  [["id", 7], ["LIMIT", 1]]
=> true
irb(main):369:0> User.exists?("917c0-bbf6-0135-c82a-1234")
  User Exists (1.0ms)  SELECT  1 AS one FROM "users" WHERE "users"."id" = $1 LIMIT $2  [["id", 917], ["LIMIT", 1]]
=> false
```

By coincidence I have a user with the id of 7, so the first one returns true when the uid doesn't actually exist.

Reply:
Understood.

Reply:
That'll be Rails type coercing into an integer. If you run `to_i` on a string it'll stop when it hits a alphanumeric character (though I be there are other complications)